browser/src/control/Control.MobileWizardWindow.js

```diff
@@ -299,13 +299,21 @@ L.Control.MobileWizardWindow = L.Control.extend({
 				this._customTitle ? this._setCustomTitle(this._customTitle) : this._setTitle(this._mainTitle);
 
 			var currentNode = $('.ui-explorable-entry.level-' + this._currentDepth + '.mobile-wizard:visible');
-			var headers = currentNode.siblings();
+			// select only those nodes which are updated on Level down
+			var headers = currentNode.siblings().filter(function() {
+				return $(this).attr('style') === 'display: none;';
+			});
 			var currentHeader = currentNode.children('.ui-header');
 			headers = headers.add(currentHeader);
 
 			var parent = currentNode.parent();
 			if (parent.hasClass('toolbox'))
-				headers = headers.add(parent.siblings());
+				parent.siblings().each(function() {
+					// select only those nodes which are updated on Level down
+					if ($(this).attr('style') === 'display: none;') {
+						headers = headers.add($(this));
```

Comment:
Is this ok? Meaning, will it always work? wouldn't be better to `$(this).style.display === 'none'`

Reply:
`$(this).style.display === 'none'` will not work as it consider all the elements which have display = none ( including the elements which we forcefully hide by css )

this condition will only consider elements which have inline property display = none

Reply:
As i am seeing you are right if in future there are addition inline css line added then this fail 🤔